lib/addons/argocd/index.ts

```diff
@@ -1,26 +1,359 @@
+            cluster: clusterInfo.cluster,
+            manifest: [{
+                apiVersion: 'v1',
+                kind: 'Namespace',
+                metadata: {
+                    name: this.options.namespace,
+                }
+            }],
+            overwrite: true,
+            prune: true
+        });
+    }
+
+    /**
+     * Creates a secret key 
+     * @param clusterInfo 
+     * @param secretName 
+     * @param dependency dependency for the created secret to control order of execution 
+     * @returns reference to the secret to add to the ArgoCD config map
+     */
+    protected async createSecretKey(clusterInfo: ClusterInfo, dependency: KubernetesManifest): Promise<string> {
+
+        const secretName = this.options.bootstrapRepo?.credentialsSecretName;
+        if (!secretName) {
+            return "";
+        }
+
+        const appRepo = this.options.bootstrapRepo!;
+        let credentials = { url: btoa(appRepo.repoUrl) };
+
+        const secretValue = await this.getSecretValue(secretName, clusterInfo.cluster.stack.region);
+
+        let result = "";
+
+        switch (appRepo?.credentialsType) {
+            case "SSH":
+                credentials = { ...credentials, ...{ sshPrivateKey: btoa(secretValue) } };
+                result = sshRepoRef(appRepo.repoUrl, secretName);
+                break;
+            case "USERNAME":
+            case "TOKEN":
+                // eslint-disable-next-line no-case-declarations
+                const secretJson: any = JSON.parse(secretValue);
+                credentials = { ...credentials, ...{
+                        username: btoa(secretJson["username"]),
+                        password: btoa(secretJson["password"])
+                    }
+                };
+                result = userNameRepoRef(appRepo.repoUrl, secretName);
+                break;
+        }
+
+        const manifest = new KubernetesManifest(clusterInfo.cluster.stack, "argo-bootstrap-secret", {
+            cluster: clusterInfo.cluster,
+            manifest: [{
+                apiVersion: "v1",
+                kind: "Secret",
+                metadata: {
+                    name: secretName,
+                    namespace: this.options.namespace,
+                    labels: {
+                        "argocd.argoproj.io/secret-type": "repo-creds"
+                    }
+                },
+                data: credentials,
+            }],
+            overwrite: true,
+        });
+
+        manifest.node.addDependency(dependency);
+        return result;
+    }
+
+    /**
+     * Gets secret value from AWS Secret Manager. Requires access rights to the secret, specified by the secretName parameter.
+     * @param secretName name of the secret to retrieve
+     * @param region 
+     * @returns 
+     */
+    protected async getSecretValue(secretName: string, region: string): Promise<string> {
```

Comment:
Same comment as below. Like a function other addons could leverage. Not specific to argo.

Reply:
I will move it out to something like secret-utils under utils.